File: paddler_client_python/paddler_client/types/conversation_message_content_part.py
from typing import Annotated, Literal

from pydantic import BaseModel, Discriminator, Tag

from paddler_client.types.image_url import ImageUrl


class TextContentPart(BaseModel):
    type: Literal["text"] = "text"
    text: str


class ImageUrlContentPart(BaseModel):
    type: Literal["image_url"] = "image_url"
    image_url: ImageUrl


ConversationMessageContentPart = Annotated[
    Annotated[TextContentPart, Tag("text")]
    | Annotated[ImageUrlContentPart, Tag("image_url")],
    Discriminator("type"),
]

File: paddler_client_javascript/src/schemas/ConversationMessageContentPart.ts
import { z } from "zod";

export const ConversationMessageContentPartSchema = z.union([
  z.object({
    type: z.literal("text"),
    text: z.string(),
  }),
  z.object({
    type: z.literal("image_url"),
    image_url: z.object({ url: z.string() }),
  }),
]);

export type ConversationMessageContentPart = z.infer<
  typeof ConversationMessageContentPartSchema
>;
